### DISCUSSION TITLE
`zppy` to run `e3sm_diags` for experiment pairs

### DISCUSSION THREAD

**Discussion by mahf708** [E3SM-Project/e3sm_diags#647] [Q&A] [answered]
Hi @forsyth2 and team,

I am wondering if there is a way compare two experiments using zppy and e3sm_diags. I was told to run standard e3sm_diags "comparison" — basically, we have an experiment pair that we are interested in looking at the difference in some outputs. Could someone please point me to the documentation and/or examples of such a routine? I tried looking around, got lost, and I thought it might be easier to just ask here.

Thank you!

**Comment by forsyth2** [comment, marked as answer]
It sounds like you want to do a model-vs-model run. Example parameters for running model-vs-model from `zppy` can be found at https://github.com/E3SM-Project/zppy/blob/main/tests/integration/generated/test_complete_run_chrysalis.cfg#L108 (the `[[ atm_monthly_180x360_aave_mvm ]]` section)

**Reply by forsyth2** [reply]
https://e3sm-project.github.io/e3sm_diags/_build/html/master/examples.html also provides a number of examples.

**Reply by mahf708** [reply]
In https://github.com/E3SM-Project/e3sm_diags/blob/main/examples/ex4-model-vs-model/diags.cfg, how do I specify the runs exactly? Is there a fully reproducible .cfg file for model-vs-model climatology?

**Reply by forsyth2** [reply]
I haven't run these in a while, but I think you'd just need to run:
```
salloc --nodes=1 --partition=debug --time=01:00:00
source /lcrc/soft/climate/e3sm-unified/load_latest_e3sm_unified_chrysalis.sh
python ex4.py --multiprocessing --num_workers=32 -d diags.cfg
chmod -R 755 <your web directory>
```

**Reply by chengzhuzhang** [reply]
In the same folder, https://github.com/E3SM-Project/e3sm_diags/blob/main/examples/ex4-model-vs-model/ex4.py is to specify the run. 
[Here](https://github.com/E3SM-Project/e3sm_diags/blob/main/e3sm_diags/driver/default_diags/lat_lon_model_vs_model.cfg) is the default set of .cfg file for lat-lon comparison,  if `run_type` is `model-vs-model`

**Reply by mahf708** [reply]
Ah yeah, I didn't realize they would go hand-in-hand. Thanks! I will try this and report back if unsuccessful. Thanks a lot. I will mark this as resolved for now :)

**Comment by mahf708** [comment]
I think that's it, but a quick follow-up to ensure I understand the logic: The idea is to compare the different years in the example you cited, right? So 1985–2014 vs 1850–1851. If I want to compare the same years, but different runs, I would simply the paths and names, right?


```cfg

[e3sm_diags]
active = True
grid = '180x360_aave'
obs_ts = "/lcrc/group/e3sm/diagnostics/observations/Atm/time-series/"
ref_final_yr = 2014
ref_start_yr = 1985
# TODO: this directory is missing OMI-MLS
reference_data_path = "/lcrc/group/e3sm/diagnostics/observations/Atm/climatology/"
sets = "lat_lon","zonal_mean_xy","zonal_mean_2d","polar","cosp_histogram","meridional_mean_2d","enso_diags","qbo","diurnal_cycle","annual_cycle_zonal_mean","streamflow", "zonal_mean_2d_stratosphere", "tc_analysis",
short_name = 'v2.LR.historical_0201'
ts_num_years = 2
walltime = "00:30:00"
years = "1850:1854:2", "1850:1854:4",

...

  [[ atm_monthly_180x360_aave_mvm ]]
  # Test model-vs-model using the same files as the reference
  climo_diurnal_frequency = "diurnal_8xdaily"
  climo_diurnal_subsection = "atm_monthly_diurnal_8xdaily_180x360_aave"
  climo_subsection = "atm_monthly_180x360_aave"
  diff_title = "Difference"
  partition = "compute"
  qos = "regular"
  ref_final_yr = 1851
  ref_name = "v2.LR.historical_0201"
  ref_start_yr = 1850
  ref_years = "1850-1851",
  reference_data_path = "/lcrc/group/e3sm/ac.forsyth2/zppy_test_complete_run_output/v2.LR.historical_0201/post/atm/180x360_aave/clim"
  run_type = "model_vs_model"
  short_ref_name = "v2.LR.historical_0201"
  swap_test_ref = False
  tag = "model_vs_model"
  ts_num_years_ref = 2
  ts_subsection = "atm_monthly_180x360_aave"
  walltime = "2:00:00"
  years = "1852-1853",
```

**Reply by forsyth2** [reply]
Yes, if you want to compare different runs, the paths need to be updated. `reference_data_path` would be changed.
